test/spec/Feature/Query/UpdateSpec.hs

```diff
@@ -535,11 +541,136 @@ spec = do
+      , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+      ]|]
+
+    it "can update tables with composite pk" $ do
+      baseTable "bulk_update_items_cpk" "id" tblDataBeforeBulk
+      `mutatesWith`
+      requestMutation methodPatch "/bulk_update_items_cpk" [("Prefer", "params=multiple-objects")]
+        [json|[
+          { "id": 1, "name": "item-1", "observation": "Lost item" }
+        , { "id": 2, "name": "item-2", "observation": null }
+        ]|]
+      `shouldMutateInto`
+      [json|[
+        { "id": 1, "name": "item-1", "observation": "Lost item" }
+      , { "id": 2, "name": "item-2", "observation": null  }
+      , { "id": 3, "name": "item-3", "observation": null  }
+      ]|]
+
+    it "updates using both the filters from the query string and the body" $ do
+      baseTable "bulk_update_items" "id" tblDataBeforeBulk
+      `mutatesWith`
+      requestMutation methodPatch "/bulk_update_items?id=gte.2" [("Prefer", "params=multiple-objects")]
+        [json|[
+          { "id": 1, "name": "item-1 - 1st", "observation": "Lost item" }
+        , { "id": 2, "name": "item-2 - 2nd", "observation": "Damaged item" }
+        , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+        ]|]
+      `shouldMutateInto`
+      [json|[
+        { "id": 1, "name": "item-1", "observation": null }
+      , { "id": 2, "name": "item-2 - 2nd", "observation": "Damaged item" }
+      , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+      ]|]
+
+--    it "updates with limit and offset" $ do
+--      baseTable "bulk_update_items" "id" tblDataBeforeBulk
+--      `mutatesWith`
+--      requestMutation methodPatch "/bulk_update_items?limit=2&offset=1&order=id" [("Prefer", "params=multiple-objects")]
+--        [json|[
+--          { "id": 1, "name": "item-1 - 1st", "observation": "Lost item" }
+--        , { "id": 2, "name": "item-2 - 2nd", "observation": "Damaged item" }
+--        , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+--        ]|]
+--      `shouldMutateInto`
+--      [json|[
+--        { "id": 1, "name": "item-1", "observation": null }
+--      , { "id": 2, "name": "item-2 - 2nd", "observation": "Damaged item" }
+--      , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+--      ]|]
+--
+--    it "updates without the header 'Prefer: params=multiple-objects', taking only the first item in the json array" $ do
+--      baseTable "bulk_update_items" "id" tblDataBeforeBulk
+--      `mutatesWith`
+--      requestMutation methodPatch "/bulk_update_items" mempty
+--        [json|[
+--          { "id": 4, "name": "item-4", "observation": "Damaged item" }
+--        , { "id": 3, "name": "item-3 - 3rd", "observation": null }
+--        ]|]
+--      `shouldMutateInto`
+--      [json|[
+--        { "id": 2, "name": "item-2", "observation": null }
+--      , { "id": 3, "name": "item-3", "observation": null }
+--      , { "id": 4, "name": "item-4", "observation": "Lost item" }
+--      ]|]
```

Comment:
I'm not sure if both bulk update and limits should be applied at the same time. Can't find a use case that may use this either. These tests fail btw, they're just an idea of how the interaction may work.

Reply:
@laurenceisla Yeah, limits and bulk update doesn't make sense IMO. With bulk update the user would have to limit the amount of array elements sent.